### DISCUSSION TITLE
[Build failed] `make CMAKE_BUILD_TYPE=RelWithDebInfo` failed

### DISCUSSION THREAD

**Discussion by AlejandroSuero** [neovim/neovim#25970] [General]
### Problem

When I used `git fetch` and then `git pull` the changes, then I run `make CMAKE_BUILD_TYPE=RelWithDebInfo` as said in the documentation.

My system is this:

<img width="392" alt="Screenshot 2023-11-10 at 14 02 05" src="https://github.com/neovim/neovim/assets/71392160/193c2289-ff80-4ae5-9d5f-fcc0bcbcd1ae" />

With all that done, when running the `make` command, outputs the following [error](https://github.com/AlejandroSuero/AlejandroSuero/blob/main/neovim-build-log.txt)

Log is too long for I got this github error `There was an error creating your Issue: body is too long (maximum is 65536 characters).`

### Steps to reproduce

```bash
git clone https://github.com/neovim/neovim
cd neovim & make CMAKE_BUILD_TYPE=RelWithDebInfo
```

### Expected behavior

To build the latest neovim nightly version.

### Neovim version (nvim -v)

0.10.0-dev-1464 commit gd3e9cbedc

### Vim (not Nvim) behaves the same?

no, vim 9.0.2049

### Operating system/version

macOS 14.0

### Terminal name/version

kitty 0.29.2

### $TERM environment variable

xterm-kitty

### Installation

build from repo

**Comment by zeertzjq** [comment]
I don't think there is anything that can be done about this.

**Comment by dundargoc** [comment]
What's the output of `cmake -E capabilities`?

**Reply by AlejandroSuero** [reply]
It outputs the following:

```json
{
  "debugger": true,
  "fileApi": {
    "requests": [
      {
        "kind": "codemodel",
        "version": [
          {
            "major": 2,
            "minor": 6
          }
        ]
      },
      {
        "kind": "configureLog",
        "version": [
          {
            "major": 1,
            "minor": 0
          }
        ]
      },
      {
        "kind": "cache",
        "version": [
          {
            "major": 2,
            "minor": 0
          }
        ]
      },
      {
        "kind": "cmakeFiles",
        "version": [
          {
            "major": 1,
            "minor": 0
          }
        ]
      },
      {
        "kind": "toolchains",
        "version": [
          {
            "major": 1,
            "minor": 0
          }
        ]
      }
    ]
  },
  "generators": [
    {
      "extraGenerators": [],
      "name": "Watcom WMake",
      "platformSupport": false,
      "toolsetSupport": false
    },
    {
      "extraGenerators": [
        "Kate"
      ],
      "name": "Ninja Multi-Config",
      "platformSupport": false,
      "toolsetSupport": false
    },
    {
      "extraGenerators": [
        "CodeBlocks",
        "CodeLite",
        "Eclipse CDT4",
        "Kate",
        "Sublime Text 2"
      ],
      "name": "Ninja",
      "platformSupport": false,
      "toolsetSupport": false
    },
    {
      "extraGenerators": [],
      "name": "Xcode",
      "platformSupport": false,
      "toolsetSupport": true
    },
    {
      "extraGenerators": [
        "CodeBlocks",
        "CodeLite",
        "Eclipse CDT4",
        "Kate",
        "Sublime Text 2"
      ],
      "name": "Unix Makefiles",
      "platformSupport": false,
      "toolsetSupport": false
    }
  ],
  "serverMode": false,
  "tls": true,
  "version": {
    "isDirty": false,
    "major": 3,
    "minor": 27,
    "patch": 7,
    "string": "3.27.7",
    "suffix": ""
  }
}
```
It may as well could be on my end, because I am trying to track a problem I have with [sketchybar](https://github.com/FelixKratz/SketchyBar) and `curl` command. The [issue](https://github.com/FelixKratz/homebrew-formulae/issues/6) if you think is related in some way.

I am a bit new to unix and building from source still, so any tips for tracking problems down would be gladly received.

**Reply by dundargoc** [reply]
Hmm, no, it looks good. How did you install cmake?

**Reply by AlejandroSuero** [reply]
I installed it with all the [build prerequisites](https://github.com/neovim/neovim/wiki/Building-Neovim#macos--homebrew) from the documentation.

```bash
brew install ninja cmake gettext curl
```

**Reply by dundargoc** [reply]
Huh. Weird. Maybe try installing neovim from homebrew as well?

**Reply by AlejandroSuero** [reply]
From homebrew seems to work fine, any way of using the nightly version from homebrew?

**Reply by clason** [reply]
No. We don't support distro nightly builds (and homebrew's in particular, which don't bundle required treesitter parsers).

You can download the nightlies from our release page, though.

**Reply by benjiwolff** [reply]
@AlejandroSuero Feel free to use my tap [benjiwolff/homebrew-neovim-nightly](https://github.com/benjiwolff/homebrew-neovim-nightly).
